src/common/transformations/src/transformations/common_optimizations/matmul_const_transposes_extraction.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/matmul_const_transposes_extraction.hpp"
+
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::MatMulConstTransposesExtraction, "MatMulConstTransposesExtraction", 0);
+
+ngraph::pass::MatMulConstTransposesExtraction::MatMulConstTransposesExtraction() {
+    auto data_pattern = pattern::any_input();
+    auto weights_pattern = pattern::wrap_type<opset8::Constant,
+                                              opset8::FakeQuantize>([](Output<Node> node) -> bool {
+                                                                        const auto& rank = node.get_partial_shape().rank();
+                                                                        return rank.is_static() && rank.get_length() >= 2;
+                                                                   });
+    auto matmul_pattern = pattern::wrap_type<opset8::MatMul>({data_pattern, weights_pattern});
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto node = m.get_match_root();
+        auto matmul = as_type<opset8::MatMul>(node.get());
+        if (!matmul || matmul->get_transpose_b())
+            return false;
+
+        const auto& pattern_value_map = m.get_pattern_value_map();
+        const auto& weights = pattern_value_map.at(weights_pattern);
+
+        std::vector<int> transpose_order(weights.get_partial_shape().size());
+        std::iota(transpose_order.begin(), transpose_order.end(), 0);
+        std::reverse(transpose_order.end() - 2, transpose_order.end());
+        auto transpose = std::make_shared<opset8::Transpose>(weights, op::Constant::create(element::i32, {transpose_order.size()}, transpose_order));
```

Comment:
I was also thinking that it is better to constant fold this Const->Transpose inplace because other passes like MatMulMultiplyFusion https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/transformations/common_optimizations/matmul_multiply_fusion.cpp#L163 can benefit from it.